mindsdb/integrations/handlers/xgboost_handler/xgboost_handler.py

```diff
@@ -0,0 +1,471 @@
+import category_encoders as ce
+from enum import Enum
+import json
+import numpy as np
+import optuna
+import pandas as pd
+import pickle
+from sklearn.metrics import accuracy_score, mean_squared_error
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import OneHotEncoder, LabelEncoder
+from xgboost import XGBClassifier, XGBRegressor
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities import log
+import xgboost as xgb
+
+
+class BoosterType(Enum):
+    gbtree = "gbtree"
+    gblinear = "gblinear"
+
+
+class ObjectiveType(Enum):
+    reg_linear = "reg:linear"
+    reg_squarederror = "reg:squarederror"
+    multi_softmax = "multi:softmax"
+    multi_softprob = "multi:softprob"
+
+
+class TaskType(Enum):
+    auto = "auto"
+    regressor = "regressor"
+    classifier = "classifier"
+    a = "a"
+    c = "c"
+    r = "r"
+    reg = "reg"
+    cla = "cla"
+
+
+def enum_to_str(type_class: Enum) -> str:
+    all = []
+    for element in type_class:
+        all.append(element.name)
+    return "|".join(all)
+
+
+def to_ts_dataframe(df: pd.DataFrame, time_col=None) -> str:
```

Comment:
From what I can see, this handler does not support time series in its current form. Do we need this method, then?